st2common/tests/unit/test_db_fields.py

```diff
@@ -73,6 +73,16 @@ class ModelWithJSONDictFieldDB(stormbase.StormFoundationDB):
 
 
 class JSONDictFieldTestCase(unittest2.TestCase):
+    def test_set_to_mongo(self):
+        field = JSONDictField(use_header=False)
+        result = field.to_mongo({"test": {1, 2}})
```

Comment:
Round trip test would also be good - aka to ensure that when we unserialize the value, we get a list back.